File: 0392.判断子序列/0392-判断子序列.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        i,j = 0,0
        while i < len(s) and j < len(t):
            while j < len(t) and t[j] != s[i]:
                j += 1
            while i < len(s) and j < len(t) and s[i] == t[j]:
                i += 1
                j += 1
        return i == len(s)

File: 0392.判断子序列/0392-判断子序列.rs
impl Solution {
    pub fn is_subsequence(s: String, t: String) -> bool {
        let s_chars: Vec<char> = s.chars().collect();
        let t_chars: Vec<char> = t.chars().collect();
        let mut i  = 0;
        let mut j = 0;
        while i < s_chars.len() && j < t_chars.len() {
            while j < t_chars.len() && t_chars[j] != s_chars[i] {
                j += 1;
            }
            while i < s_chars.len() && j < t_chars.len() && s_chars[i] == t_chars[j] {
                i += 1;
                j += 1;
            }
        }
        
        i == s_chars.len()
    }
}